FinniversKit/Sources/Components/BrazePromotionView/BrazePromotionView.swift

```diff
@@ -0,0 +1,249 @@
+        button.layer.masksToBounds = true
+        button.layer.cornerRadius = buttonSize / 2.0
+        button.clipsToBounds = true
+        button.imageEdgeInsets = UIEdgeInsets(vertical: .spacingXS, horizontal: .spacingXS)
+        button.addTarget(self, action: #selector(handleTapOnCloseButton), for: .touchUpInside)
+        return button
+    }()
+
+    private lazy var remoteImageView: RemoteImageView = {
+        let imageView = RemoteImageView(withAutoLayout: true)
+        imageView.image = UIImage(named: ImageAsset.noImage)
+        imageView.contentMode = .scaleAspectFill
+        imageView.layer.masksToBounds = true
+        imageView.clipsToBounds = true
+        return imageView
+    }()
+
+    private lazy var verticalStackView: UIStackView = {
+        let stackView = UIStackView(axis: .vertical, spacing: .spacingS + .spacingXS, withAutoLayout: true)
+        stackView.distribution = .fillProportionally
+        stackView.alignment = .leading
+        stackView.setContentCompressionResistancePriority(.required, for: .horizontal)
+        return stackView
+    }()
+
+    private lazy var stackViewConstraintsImage: [NSLayoutConstraint] = [
+        verticalStackView.topAnchor.constraint(equalTo: backgroundView.topAnchor, constant: .spacingM),
+        verticalStackView.leadingAnchor.constraint(equalTo: backgroundView.leadingAnchor, constant: .spacingM),
+        verticalStackView.bottomAnchor.constraint(equalTo: backgroundView.bottomAnchor, constant: -.spacingM),
+        verticalStackView.widthAnchor.constraint(equalTo: widthAnchor, multiplier: 0.6),
+    ]
+
+    private lazy var stackViewConstraintsNoImage: [NSLayoutConstraint] = [
+        verticalStackView.topAnchor.constraint(equalTo: backgroundView.topAnchor, constant: .spacingM),
+        verticalStackView.leadingAnchor.constraint(equalTo: backgroundView.leadingAnchor, constant: .spacingM),
+        verticalStackView.bottomAnchor.constraint(equalTo: backgroundView.bottomAnchor, constant: -.spacingM),
+        verticalStackView.trailingAnchor.constraint(equalTo: backgroundView.trailingAnchor, constant: -.spacingM),
+    ]
+
+    private var viewModel: BrazePromotionViewModel
+    private var imageDatasource: RemoteImageViewDataSource?
+
+    // MARK: - Public properties
+
+    public enum Action {
+        case primary
+        case secondary
+    }
+
+    public weak var delegate: BrazePromotionViewDelegate?
+
+    // MARK: - Init
+
+    public init(viewModel: BrazePromotionViewModel, imageDatasource: RemoteImageViewDataSource) {
+        self.viewModel = viewModel
+        self.imageDatasource = imageDatasource
+        super.init(frame: .zero)
+        setup()
+        configure(with: viewModel, imageDatasource: imageDatasource)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    public func configure(with viewModel: BrazePromotionViewModel, imageDatasource datasource: RemoteImageViewDataSource) {
```

Comment:
The parameter `imageDatasource` isn't used in this method, but the method is public.

If the view should be reconfigurable we should set `self.imageDatasource = datasource` from this method, if not the parameter should be removed.

Reply:
Same with `viewModel`. It's used, but it's not saved

Reply:
I actually hadn't decided yet when I wrote this, which is why it reads a bit confused 😆 Yeah, I don't think it needs to be reconfigurable because it will be replaced quite rarely (I think). Any other opinions please let me know, I'm open to other perspectives on this.

Reply:
I will make them non-reconfigurable after discussion with @ninarg